### DISCUSSION TITLE
Older Versions Lurking on Github...

### DISCUSSION THREAD

**Discussion by survivethewalkingdead** [fra589/grbl-Mega-5X#251] [Q&A] [answered]
fra589,
 I do not remember exactly where I went to download grbl-Mega-5x-edge, but that is the name, no doubt I got here (github) from google, on 1 Feb 2022. It is an older(?) version... for example it has over 560 lines in defaults.h and Y is cloned on line 556-557. The one I just downloaded has just over 100 lines and A and B axis are enabled. Sorry I don't have more details for you, but thought you should know. I was browsing various conversations and one between you and Lesli sounded like what I was seeing...default axis 4 cloned to axis 2. I went back to my untouched downloaded file and it was in there. Came here, downloaded again, extracted, way fewer lines in defaults.h, installed, put on my MKS Gen 2Z V1.1,Opened Cn5X and voila! Doing everything I tried to enable for last 5 hrs! 
I have the original downloaded zip file if you need it. 
Once again, thanks for all you do!
Darren

**Comment by fra589** [comment]
Hi @survivethewalkingdead,  
  
What you say is strange...  
There was never any axis cloning in the default versions of grbl-Mega-5X.  
Also, defining axis cloning is done in config.h and not in default.h.  
The cleanup of unnecessary portions of code dates from April 29, 2021. It is this cleanup that led to the deletion of 425 lines in the default.h file.  
I really want you to send me your origina zip to see the differences with my versions, but I don't think it comes from here, or there was a problem during the transfer.  
  
@++;  
Gauthier.

**Comment by survivethewalkingdead** [comment, marked as answer]
This is the exact file downloaded from github on 1 feb 2022. I may not have
clicked the download zip on the actual intended main Mega-5X page. Like I
said, I learned of it from Google, clicked the links it brought up, and
here we are. Maybe google took me to an archived site? I have no concrete,
constructive idea. I mostly mention it so that you are aware that some of
the issues you are replying to are related to incorrect source
files...example: Issue 156, 26Nov2020, I'm sure Lesli had the same file I
am attaching here.

Thanks again,
Darren

**Reply by fra589** [reply]
In fact, it was an old version of config.h file with special settings, it was not downloaded from the main Github's fra589/grbl-Mega-5X page.

**Reply by survivethewalkingdead** [reply]
Yup. I don't know if Google archives old links or what, but I have since learned to make sure I am on the correct page, use github links to move around and not just download the first file that shows up from a Google search. :)

On another note, I have managed to break it again...I assume it it is the grbl-Mega-5x, my mega, my computer or the gcode file. The problem shouldn't be cn5x because it also crashes grbl-gru and grbl panel. Sorry, I don't have any details right now, I'll play with it again on Wed. 

Darren.

**Reply by survivethewalkingdead** [reply]
Last week my gcode ran to line 38, program would hang briefly and then close. This week it would pause and have jog out of table size error. Once I enlarged the table definition to have enough room, program ran. Still something weird...the green pulsing bar, I think it is supposed to represent the buffer, or read ahead, or something? When I was first playing with program, that pulsed steadily at around 25%. Now it is blank most of the time until I click through a bunch of set origine options and then suddenly it will be back. I haven't figured out what exactly I have to click to wake it up... 
sample gcode I was running on bare mega2560 with unmodified mega-5x... 

 G90
 G20
 G0 X5.1066 Y3.8387
 M3
 G4 P1.
 G1 X5.1548 Y3.9021 F150.
 G1 X4.944 Y3.8515
 G3 X4.8552 Y3.7069 I0.028 J-0.1167
 G3 X5.2248 Y3.7954 I0.1848 J0.0443 F100.
 G3 X4.8552 Y3.7069 I-0.1848 J-0.0443
 M5

 G1 X4.4172 Y3.6019 F300.
 G1 X5.1442 Y0.7241
 M3
 G4 P1.
 G1 X5.2196 Y0.7498 F100.
 G1 X5.0164 Y0.8255
 G3 X4.8621 Y0.755 I-0.0419 J-0.1124
 G3 X5.2181 Y0.6223 I0.178 J-0.0664 F100.
 G3 X4.8621 Y0.755 I-0.178 J0.0664
 M5

 G1 X4.44 Y0.9124 F300.
 G1 X0.7259 Y0.6173
 M3
 G4 P1.
 G1 X0.7926 Y0.5737 F100.
 G1 X0.7271 Y0.7804
 G3 X0.5765 Y0.8586 I-0.1144 J-0.0362
 G3 X0.6912 Y0.4964 I0.0574 J-0.1811 F100.
 G3 X0.5765 Y0.8586 I-0.0574 J0.1811
 M5

 G1 X0.4405 Y1.288 F300.
 G1 X0.6096 Y3.8473
 M3
 G4 P1.
 G1 X0.5921 Y3.925 F100.
 G1 X0.4951 Y3.7311
 G3 X0.5487 Y3.5701 I0.1073 J-0.0537
 G3 X0.7188 Y3.9099 I0.085 J0.1699 F100.
 G3 X0.5487 Y3.5701 I-0.085 J-0.1699
 M5

 G1 X0.3472 Y3.1673 F300.
 G1 X-0.3369 Y2.4331
 M3
 G4 P1.
 G1 X-0.12 F100.
 G3 X0. Y2.5531 I0. J0.12
 G1 Y3.8025 F100.
 G2 X0.59 Y4.3925 I0.59 J0.
 G1 X5.24
 G2 X5.83 Y3.8025 I0. J-0.59
 G1 Y0.59
 G2 X5.24 Y0. I-0.59 J0.
 G1 X0.59
 G2 X0. Y0.59 I0. J0.59
 G1 Y2.5531
 M5

 M30
